kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+
+int sem_destroy(sem_t *semaphore)
+{
+	k_sem_reset(semaphore->sem);
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	int count;
+
+	count = k_sem_count_get(semaphore->sem);
+	*value = count;
+
+	return 0;
+}
+
+int sem_init(sem_t *semaphore, int pshared, unsigned int value)
+{
+	if ((semaphore->sem->limit) == 0) {
+		return -1;
+	}
+	ARG_UNUSED(pshared);
```

Comment:
always put ARG_UNUSED() at the beginning (after var declaration, if any)

Reply:
For compatibility APIs, I would prefer to see an ASSERT() for parameters.  This is to reduce surprises from users porting their code that might be relying on parameters in a way that's different than what we're assuming from this shim implementation.